extensions/websockets-next/runtime/src/main/java/io/quarkus/websockets/next/runtime/WebSocketConnectionBase.java

```diff
@@ -131,7 +131,12 @@ public BroadcastSender broadcast() {
     public CloseReason closeReason() {
         WebSocketBase ws = webSocket();
         if (ws.isClosed()) {
-            return new CloseReason(ws.closeStatusCode(), ws.closeReason());
+            Short code = ws.closeStatusCode();
+            if (code == null) {
+                // This could happen if the connection is terminated abruptly
+                return CloseReason.INTERNAL_SERVER_ERROR;
```

Comment:
Does it make sense to return internal server error in this case?

Reply:
Well, it's kind of an internal server error. Also I did not find a better code in the list defined by the spec: https://datatracker.ietf.org/doc/html/rfc6455#section-7.4.1. There is `1006` but it's reserved and has a special meaning.